hw-36.ts

```diff
@@ -0,0 +1,180 @@
+        country: "United States, United Kingdom",
+        poster: "https://m.media-amazon.com/images/M/MV5BNzVlY2MwMjktM2E4OS00Y2Y3LWE3ZjctYzhkZGM3YzA1ZWM2XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg",
+        imdbRating: 8.6,
+        imdbVotes: 1259440,
+        type: "movie",
+        boxOffice: "$460,998,507",
+        production: "Lucasfilm Ltd.",
+    },
+    {
+        id: 4,
+        title: "Harry Potter and the Half-Blood Prince",
+        year: 2009,
+        released: "15 Jul 2009",
+        runtime: "153 min",
+        genre: ["Action", "Adventure", "Family"],
+        director: "David Yates",
+        writer: "Steve Kloves, J.K. Rowling",
+        actors: ["Daniel Radcliffe", "Emma Watson", "Rupert Grint"],
+        plot: "As Harry Potter begins his sixth year at Hogwarts, he discovers an old book marked as 'the property of the Half-Blood Prince' and begins to learn more about Lord Voldemort\'s dark past.",
+        country: "United Kingdom",
+        poster: "https://m.media-amazon.com/images/M/MV5BNzU3NDg4NTAyNV5BMl5BanBnXkFtZTcwOTg2ODg1Mg@@._V1_SX300.jpg",
+        imdbRating: 7.6,
+        imdbVotes: 492245,
+        boxOffice: "$302,305,431",
+        production: "Heyday Films, Warner Bros.",
+    },
+    {
+        id: 5,
+        title: "Harry Potter and the Sorcerer's Stone",
+        year: 2001,
+        released: "16 Nov 2001",
+        runtime: "152 min",
+        genre: ["Adventure", "Family", "Fantasy"],
+        director: "Chris Columbus",
+        writer: "J.K. Rowling, Steve Kloves",
+        actors: ["Daniel Radcliffe", "Rupert Grint", "Richard Harris"],
+        plot: "An orphaned boy enrolls in a school of wizardry, where he learns the truth about himself, his family and the terrible evil that haunts the magical world.",
+        country: "United Kingdom, United States",
+        poster: "https://m.media-amazon.com/images/M/MV5BNjQ3NWNlNmQtMTE5ZS00MDdmLTlkZjUtZTBlM2UxMGFiMTU3XkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg",
+        imdbRating: 7.6,
+        imdbVotes: 684604,
+        boxOffice: "$318,087,620",
+        production: "1492 Pictures, Heyday Films, Warner Brothers",
+    }
+
+]
+
+
+//Task 1
+
+const getGenre = (movies: Movies[]) => {
+    const genreArr = movies.reduce((acc, movie) => [...acc, ...movie.genre], [])
+    return Array.from(new Set(genreArr))
+}
+
+console.log('Genres:', getGenre(movies))
+
+//Task 2
+
+const getActors = (movies: Movies[]) => {
+    const actorsArr = movies.reduce((acc, movie) => [...acc, ...movie.actors], [])
+    return Array.from(new Set(actorsArr))
+}
+
+console.log('Actors:', getActors(movies))
+
+//Task 3
+
+const sortMovies = (movies: Movies[]) => {
+    const moviesArrCopy = [...movies]
+    return moviesArrCopy.sort((a, b) => a.imdbRating > b.imdbRating ? -1 : 1)
+}
+
+console.log(sortMovies(movies))
+
+//Task 4
+
+const createNewFilmArray = (movies: Movies[]) => movies.map(movie => 
```

Comment:
super!
my favorite way to do it
`movies.map(({id, title, released, plot})=>({id, title, released, plot}))`